src/dealWizard/openDealWizard/openDealWizard.ts

```diff
@@ -0,0 +1,36 @@
+import { autoinject } from "aurelia-framework";
+import { PLATFORM } from "aurelia-pal";
+import { Router, RouterConfiguration } from "aurelia-router";
+import { IWizardStage } from "dealWizard/dealWizard.types";
+import { OpenDealWizardResult } from "./openDealWizardResult";
+import { DealWizardService } from "../../services/DealWizardService";
+
+@autoinject
+export class OpenDealWizard {
+  public wizardManager = this;
+  private stages: IWizardStage[] = [{
+    name: "Proposal",
+    valid: false,
+    route: "stage1",
+    moduleId: PLATFORM.moduleName("../stages/proposalStage/proposalStage"),
+  }, {
+    name: "Primary DAO",
+    valid: false,
+    route: "stage2",
+    moduleId: PLATFORM.moduleName("../stages/primaryDao/primaryDao"),
+  }];
+
+  private wizardResult = new OpenDealWizardResult();
+
+  constructor(private dealWizardService: DealWizardService) {
+    this.dealWizardService.registerWizard(this, this.stages, this.wizardResult);
+  }
+
+  private attached(): void {
+    this.dealWizardService.updateIndexOfActiveBaseOnRoute(this);
```

Comment:
doesn't this have to be called when attaching to each stage?

Reply:
we should maintain this state in the service.  I think we can do this in the routing logic.